student/src/main/java/tds/student/tdslogger/ApplicationStartupLogger.java

```diff
@@ -0,0 +1,31 @@
+package tds.student.tdslogger;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.stereotype.Component;
+
+import static tds.student.tdslogger.StudentEventLogger.StudentLogEvent.APP_STARTUP;
+
+@Component
+public class ApplicationStartupLogger implements ApplicationListener<ContextRefreshedEvent> {
+
+  private StudentEventLogger logger;
+  boolean logged = false;
+
+  @Autowired
+  public ApplicationStartupLogger(@Qualifier("integrationObjectMapper") final ObjectMapper objectMapper) {
+    logger = new StudentEventLogger(objectMapper);
+  }
+
+  @Override
+  public void onApplicationEvent(final ContextRefreshedEvent event) {
```

Comment:
Yes. It is fired when properties, xml or any schema files are loaded/refreshed. I'm using it to log the initial load at startup, which is a common approach in spring 3.